code/entities/chair/CinemaChair.cs

```diff
@@ -0,0 +1,214 @@
+﻿using Editor;
+using Sandbox;
+using Sandbox.Diagnostics;
+using System.Collections.Generic;
+
+namespace Cinema;
+
+[Library("ent_cinemachair"), HammerEntity]
+[Model(Archetypes = ModelArchetype.animated_model, MaterialGroup = "Deep_Green_Suede", Model = "models/cinemachair/cinemachair01.vmdl")]
+[Title("Cinema Chair"), Category("Gameplay"), Icon("event_seat")]
+public partial class CinemaChair : AnimatedEntity, ICinemaUse
+{
+    /// <summary>
+    /// Returns true if a player is currently seated in this chair.
+    /// </summary>
+    public bool IsOccupied => Occupant.IsValid();
+
+    /// <summary>
+    /// The player who is currently seated in this chair, or null if no player 
+    /// is currently seated here.
+    /// </summary>
+    [Net]
+    public Player Occupant { get; protected set; }
+
+    /// <summary>
+    /// Allows access to both of the armrests associated with this chair.
+    /// </summary>
+    [Net]
+    public IDictionary<Armrest.Sides, Armrest> Armrests { get; set; }
+
+    public Armrest LeftArmrest
```

Comment:
These could cause errors on the client while the Dictionary is propagating through the network.

If you try to get a value from a dictionary with `[]` and the key doesn't exist, an exception is thrown.
Instead you can do this:
```
public Armrest LeftArmrest
        => Armrests.TryGetValue(Armrest.Sides.Left, out Armrest value) ? value : null;
```
And null will be returned instead.

Reply:
Implemented the proposed change and added various null checks for callers of Armrests, LeftArmrest, and RightArmrest.